src/Arcus.Templates.Tests.Integration/Health/HealthEndpointService.cs

```diff
@@ -39,11 +49,23 @@ public HealthEndpointService(TestConfig configuration, ITestOutputHelper outputW
         /// </summary>
         public async Task<HttpResponseMessage> GetAsync()
         {
-            _outputWriter.WriteLine("GET {0} ->", _healthEndpoint);
-            var response = await HttpClient.GetAsync(_healthEndpoint);
-            _outputWriter.WriteLine("<- {0}", response.StatusCode);
+            return await GetAsync(request => { });
+        }
+
+        /// <summary>
+        /// Sends a GET request to the health endpoint of the API web application.
+        /// </summary>
+        public async Task<HttpResponseMessage> GetAsync(Action<HttpRequestMessage> alterRequest)
+        {
+            _outputWriter.WriteLine("GET -> {0}", _healthEndpoint);
+            using (var request = new HttpRequestMessage(HttpMethod.Get, _healthEndpoint))
+            {
+                alterRequest(request);
```

Comment:
Maybe better to call a virtual method that those can override in case they have to, rather than passing an action?

Reply:
But the test case (ex. shared access key auth) determines how the request should be altered. Therefore an action can be passed allong and the test reader can see in what way the request should be changed. No?

Otherwise we have to create an seperate child class for every altered request. See?

Reply:
Overload to pass along headers instead of action?

Reply:
No that is ok, makes sense!